src/DynamoCoreWpf/Utilities/LibraryDragAndDrop.cs

```diff
@@ -29,18 +29,24 @@ internal void HandleMouseDown(Point position, NodeSearchElementViewModel node)
 
         internal void HandleMouseMove(FrameworkElement sender, Point currentPosition)
         {
-            if (isDragging || nodeViewModel == null)
+            if (isDragging)
                 return;
 
             // If item was dragged enough, then fire DoDragDrop. 
             // Otherwise it means user click on item and there is no need to fire DoDragDrop.
             var deltaX = System.Math.Abs(currentPosition.X - startPosition.X);
             if (deltaX < SystemParameters.MinimumHorizontalDragDistance)
+            {
+                isDragging = false;
                 return;
+            }
 
             var deltaY = System.Math.Abs(currentPosition.Y - startPosition.Y);
             if (deltaY < SystemParameters.MinimumVerticalDragDistance)
+            {
+                isDragging = false;
                 return;
+            }
```

Comment:
If I intend to drag-drop an item, and then it happens that the first mouse-move is too small (within `MinimumVerticalDragDistance`), then the drag operation will be canceled right away. That might not be desirable.

I think we should instead fix this with a new `HandleMouseUp` message, but that means we have to capture the mouse input when the drag begins (i.e. when we set `isDragging = true`), and then call `ReleaseMouse` in `HandleMouseUp`. Then again that brings another question: what if the caller (of `LibraryDragAndDrop`) did not call `HandleMouseUp`? That will result in hanging issue.

I suggest you have a discussion with @aosyatnik on this one, it doesn't look right to be setting `isDragging = false` here, it needs to be actively canceled by the caller.